tests/integration/piv_test.py

```diff
@@ -1388,25 +1420,244 @@ def test_piv_added_stop_time_at_the_end(mock_rabbitmq):
 
 
 def test_piv_added_stop_time_by_event(mock_rabbitmq):
+    """
+    stop is inserted on rang 3
+    by event notification instead of statusModification
+    """
     piv_23187_addition = _get_stomp_20201022_23187_stop_time_added_in_the_middle_fixture()
     modification = ModificationTuple(statut=None, motif=None)
     disruption = DisruptionTuple(type="CREATION", texte=None)
+    _set_event_on_stops(
+        fixture=piv_23187_addition,
+        rang_min=3,
+        rang_max=3,
+        disruption=disruption,
+        modification=modification,
+    )
+
+    res = api_post("/piv/{}".format(PIV_CONTRIBUTOR_ID), data=ujson.dumps(piv_23187_addition))
+    assert "PIV feed processed" in res.get("message")
+
+    _assert_db_added_stop_time_in_the_middle()
+
+
+def test_piv_re_routed_at_the_beginning(mock_rabbitmq):
+    """
+    old trip
+       "85010231" - "85010157" - "85010140" - "85162735" - "85162750" - "87745497"
+
+    new trip
+    "85010116" - "85010082"
+                      \\
+                    "85010157" - "85010140" - "85162735" - "85162750" - "87745497"
```

Comment:
Easier to understand when updates are after (below) IMO:
```suggestion
                    "85010157" - "85010140" - "85162735" - "85162750" - "87745497"
                      //
    "85010116" - "85010082"
```